[IV-23-24] Objetivo 2

smallPingu opened the pull request:
Sobre la estructura del repositorio

* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?

Sobre el planteamiento

* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

Sobre el análisis del problema

* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

Sobre el código

* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

smallPingu committed:
Añadido src/tarea.ts...

smallPingu committed:
Añadido archivo iv.yaml

smallPingu committed:
Actualizado parámetros de tarea

DarckMonster commented:
Hola! 
Voy a revisar tu objetivo 2. A primera vista te aconsejo que cambies el título ya que tiene el que trae por defecto, sería mas descriptivo `[IV-23-24] Objetivo 2` por ejemplo.

smallPingu commented:
> Hola! Voy a revisar tu objetivo 2. A primera vista te aconsejo que cambies el título ya que tiene el que trae por defecto, sería mas descriptivo `[IV-23-24] Objetivo 2` por ejemplo.

Perfecto, lo he cambiado. Gracias por la recomendación.

JJ commented:
Sería conveniente que @AlvaroSanPal95 aprobara explícitamente, o hiciera comentarios al código, antes de que yo lo aprobara. También si puede ser esta misma mañana.

JJ commented:
Realmente sería conveniente que corrigiérais esto lo antes posible, ayer mejor que hoy.

DarckMonster commented:
> Realmente sería conveniente que corrigiérais esto lo antes posible, ayer mejor que hoy.

Sí, disculpe, es que creía que no había terminado de entregarlo para revisar. Voy a remarcar los problemas que veo.

JJ commented:
> > Realmente sería conveniente que corrigiérais esto lo antes posible, ayer mejor que hoy.
> 
> Sí, disculpe, es que creía que no había terminado de entregarlo para revisar. Voy a remarcar los problemas que veo.

No lo decía por ti, pero gracias de todas formas.

DarckMonster commented:
> > > Realmente sería conveniente que corrigiérais esto lo antes posible, ayer mejor que hoy.
> > 
> > 
> > Sí, disculpe, es que creía que no había terminado de entregarlo para revisar. Voy a remarcar los problemas que veo.
> 
> No lo decía por ti, pero gracias de todas formas.

Si, ahora lo veo. Aunque voy a hacer una review ya que veo algunos problemas.

DarckMonster reviewed (commented):
Voy a empezar diciendo que en cuanto a los issues lo veo bien, sin embargo veo otros problemas:
* A primera vista falta un archivo ``.gitignore`` adecuado al lenguaje de programación.
* Revisando los issues vi que decías que habrá un atributo *Urgencia* y este no aparece en el código.
* En cuanto a los commits no siguen las directrices, no hacen referencia a ningún issue. Los commits deben referenciar el issue sobre el que están trabajando. Esto se hace mencionando la numeración del issue (#n) en el mensaje de commit.

smallPingu committed:
Update tarea.ts

Añadido enum Urgencia y variable urgencia de tipo enum urgencia

Además, para seguir la convención usada en TypeScript, se ha cambiado todo a camelCase.

smallPingu committed:
Update .gitignore

Añadida directivas correspondientes a TypeScript

smallPingu commented:
@DarckMonster He tenido en cuenta los dos primeros puntos que has comentado e intentaré seguir el tercero para el transcurso del proyecto, gracias.
Revisa una vez más y verifica si todo está bien por favor.

DarckMonster commented:
> @DarckMonster He tenido en cuenta los dos primeros puntos que has comentado e intentaré seguir el tercero para el transcurso del proyecto, gracias. Revisa una vez más y verifica si todo está bien por favor.

Bien, como el último punto es referido a los commits, tendría que haberlo hecho con los commits que acabas de hacer, intenta hacerlo de ahora en adelante.

En cuanto a los otros puntos está bien. Resuelve el comentario por favor. 

Buen trabajo :+1:

AlvaroSanPal95 commented:
Buenas @smallPingu;
Creo que deberías mencionar en los mensajes de commit los issues que cierran esos cambios de código, para establecer bien las referencias.
A parte de eso, veo correcto el contenido del PR y las issues creadas a partir de las HUs.

Buenas @JJ, tiene mi aprobación, es una buena base a partir de la que seguir trabajando y construyendo los siguientes objetivos.

JJ commented on src/tarea.ts:
```diff
@@ -0,0 +1,13 @@
+    MuyAlta
+}
+
+interface Tarea {
```
Este código sigue teniendo como mensaje de commit "update tarea.ts".

JJ replied:
Y aquí ya te lo dije.
También es el mensaje por omisión que se pone cuando se edita desde la web. En el objetivo 2 es conveniente que empecéis a manejar con soltura un IDE, y trabajar con git desde la línea de órdenes.

JJ reviewed (changes requested):
En este objetivo se trata de que sigáis una metodología, que debería ser DDD, para poder modelizar el problema. Esa metodología queda plasmada en una serie de buenas prácticas: de las HUs tienes que sacar issues (que planteen problemas) y a partir de ellos, mencionándolos en los mensajes de commit, crear el código. Si los mensajes de commit no mencionan ningún issue, es imposible saber si aporta valor al cliente (cuyas necesidades están plasmadas en las historias de usuario); de la misma forma, si no se sabe qué problema están resolviendo, el código es simplemente incorrecto. Ni siquiera se puede decir si es incompleto o no. Puede ser incompleto, completo o simplemente no resolver ningún problema de los planteados. La forma correcta de proceder es borrar el código y empezar a desarrollar desde cero, con los issues correspondientes.
Pasaré ahora a ver qué se ha hecho en tema de issues, en todo caso. En general, es bastante complicado que sean correctos si el código no ha seguido las mejores prácticas.
Adicionalmente, `src` como lugar para el código es una elección específica que no tiene por qué ser la adecuada. Hay otros nombres de directorio que se consideran más aceptables.

smallPingu committed:
Delete src/tarea.ts

Para una buena práctica, vamos a realizar el código desde 0

smallPingu committed:
Create tarea.ts

Creamos el fichero tarea.ts que usará el lenguaje de TypeScript

smallPingu committed:
Update tarea.ts

Para resolver @AlvaroSanPal95/AssignedOrganizaiton#2, vamos a crear un tipo de dato llamado Tarea.

smallPingu committed:
Update tarea.ts

Para resolver @AlvaroSanPal95/AssignedOrganization#7, vamos a crear una variable llamada Urgencia que será de tipo urgencia, un enumerado que crearemos.

JJ commented:
Arturo, "update tarea.ts" no es un mensaje de commit que siga las buenas prácticas. Prefiero que no lo sigas haciendo a tener que decírtelo cuando solicites evaluación. Todos los mensajes de commit tienen que referirse a un issue, en el que se describa un problema relacionado con las historias de usuario.

smallPingu committed:
Delete src/tarea.ts

Borramos archivo tarea.ts por un mal uso de los mensajes sin un claro propósito en los commits

smallPingu committed:
Creamos archivo tarea.ts de tipo de TypeScript

smallPingu committed:
Añadimos al archivo la interfaz Tarea para tratar con cada tarea

smallPingu commented:
> Arturo, "update tarea.ts" no es un mensaje de commit que siga las buenas prácticas. Prefiero que no lo sigas haciendo a tener que decírtelo cuando solicites evaluación. Todos los mensajes de commit tienen que referirse a un issue, en el que se describa un problema relacionado con las historias de usuario.

Entendido, perdón por la repetida confusión.

Además, he terminado con los exámenes y trabajos de las demás asignatura, y quiero ponerme a trabajar en esto al máximo 👍

JJ commented:
> > Arturo, "update tarea.ts" no es un mensaje de commit que siga las buenas prácticas. Prefiero que no lo sigas haciendo a tener que decírtelo cuando solicites evaluación. Todos los mensajes de commit tienen que referirse a un issue, en el que se describa un problema relacionado con las historias de usuario.
> 
> Entendido, perdón por la repetida confusión.
> 
> Además, he terminado con los exámenes y trabajos de las demás asignatura, y quiero ponerme a trabajar en esto al máximo 👍

Ánimo.

JJ commented on src/tarea.ts:
```diff
@@ -0,0 +1,14 @@
+  MuyAlta
+}
+
+interface Tarea 
```
Una vez más, te agradecería que pusieras en práctica las buenas prácticas en desarrollo, y que se reflejaran en mensajes de commit. Este fichero lo has creado en este commit [aa0d1c0](https://github.com/AlvaroSanPal95/AssignedOrganization/commit/aa0d1c05dabb67dac53465133a1e0c5fc263ddbb) que no solo no hace referencia a ningún issue, sino que tampoco menciona `Urgencia`. En desarrollo ágil hay que añadir valor al cliente cada vez que se crea código o se modifica, y eso se refleja en hacer referencias a los issues que a su vez deben referenciar la historia de usuario.

JJ reviewed (changes requested):
En todos los objetivos se trata de que uséis una metodología adecuada para poder resolver el problema del cliente. No se trata de escribir código, se trata de entende qué necesita el cliente, crear código que contribuya a resolver lo que se ha entendido, y hacerlo todo ello de forma comprensible en el futuro para quien lo escribe y en el presente para quien lo revisa. 
Una vez más, el código es incorrecto y la forma correcta de proceder es borrarlo y seguir la metodología correcta para crearlo de nuevo.

smallPingu committed:
Eliminamos archivo src/tarea.ts para seguir una buena práctica del desarrollo ágil

smallPingu committed:
Creamos archivo tarea.ts que contendrá una clase que podrá definir a qué nos referimos con una tarea AlvaroSanPal95/AssignedOrganization#6

smallPingu committed:
Añadimos atributos que nos ayudaran a definir cada tarea. Closes AlvaroSanPal95/AssignedOrganization#6

smallPingu committed:
Añadimos variable urgencia para determinar que tan importante es una tarea. Closes AlvaroSanPal95/AssignedOrganization#7

smallPingu commented:
@JJ 

¿Para pedir revisión le menciono en el PR?

JJ commented:
Técnicamente, deberías pedir revisión también a @AlvaroSanPal95 aunque ya está aprobado por su parte. En todo caso, la forma de pedirlo es precisamente pedir una revisión usando la opción del menú que hay arriba a la derecha. Mencionando, como ves, también lo veo.

JJ commented:
Los mensajes de commit son mejores que antes, pero deben seguir el formato 50 caracteres + retorno de carro + líneas de 80 caracters. Los primeros caracteres deben enlazar el issue. 
![Captura de pantalla de 2024-01-21 20-30-14](https://github.com/AlvaroSanPal95/AssignedOrganization/assets/500/4424a45b-46ca-43b6-861e-ca9d56b18cfe)
Si no, pasa lo de arriba. Tienes que pinchar en el commit para ver si se enlaza algún issue y si es correcto.

JJ commented on src/tarea.ts:
```diff
@@ -0,0 +1,14 @@
+  MuyAlta
+}
+
+interface Tarea
```
Para empezar, has vuelto a añadir el mismo código. Si es incorrecto, es incorrecto, y por muchas cosas que hagas sigue siendo incorrecto. La metodología hay que seguirla para llegar a un resultado currecto, no porque yo lo diga. Es conveniente que entiendas esto, porque si no el tiempo empleado por los dos no va  encaminarse a que aprendas las metodologías que se implican en cada uno de los objetivos.

JJ commented on src/tarea.ts:
```diff
@@ -0,0 +1,14 @@
+enum Urgencia {
```
En #7, para empezar, no dice absolutamente nada de qué es una tarea urgente, ni qué significa la urgencia, ni cómo va usarse en la lógica de negocio.
Es imposible llegar de #7 a 4 niveles de urgencia, y no simplemente a un valor booleano, o a un valor numérico en vez de un enum. ¿Se va a ordenar por urgencia? ¿Se van a hacer o no dependiendo de la urgencia? ¿Pueden dos tareas tener la misma urgencia? En cada uno de los casos la solución será diferente, pero si no sacas directamente un issue de la lógica de negocio, sino que lo relacionas con otro issue, es imposible acabar con un código que aporte valor al cliente.

JJ reviewed (changes requested):
Ya te comenté en los issues #6 y #7 los problemas que había. Estás creando issues como el #7, hablando de la "urgencia", que ni siquiera se menciona en la HU #2.
Esta HU tiene [problemas](https://github.com/AlvaroSanPal95/AssignedOrganization/issues/2#issuecomment-1861098143), y @AlvaroSanPal95 debería haberlos solucionado; sobre todo eliminar toda posible referencia a detalles de implementación. Pero si obviamos esos problemas, y nos quedamos con los tres primeros párrafos, o más bien el segundo y el 3º, se deberían extraer al menos un issue. Lo suyo sería que cerraras los issues incorrectos con una etiqueta "wontfix", "cerrándolos" desde un mensaje de commit en el que *borres* el código creado hasta el momento.
A continuación, tienes que crear diferentes issues *todos los cuales deben enlazar la HU* que planteen los diferentes problemas existentes.
Lo mejor sería que no hicieras nada hasta que escribas estos issues. Si vuelves a hacer código a partir de issues incorrectos, no será evaluable porque no se podrá saber si el código aporta valor o no.
En todo caso, y a menos que @AlvaroSanPal95 vaya a continuar trabajando en tu repositorio, dado que tienes que empezar de nuevo yo diría que trabajaras sobre tu propio repositorio, porque cuando llegues al objetivo 4 vas a necesitar el código del objetivo 2, y si no lo han hecho, vas a tener que hacerlo tú desde cero. Así que si @AlvaroSanPal95 no lo soluciona prácticamente en un día, yo te diría que consideraras la opción de trabajar sobre tu propio repo.

smallPingu commented:
> Ya te comenté en los issues #6 y #7 los problemas que había. Estás creando issues como el #7, hablando de la "urgencia", que ni siquiera se menciona en la HU #2. Esta HU tiene [problemas](https://github.com/AlvaroSanPal95/AssignedOrganization/issues/2#issuecomment-1861098143), y @AlvaroSanPal95 debería haberlos solucionado; sobre todo eliminar toda posible referencia a detalles de implementación. Pero si obviamos esos problemas, y nos quedamos con los tres primeros párrafos, o más bien el segundo y el 3º, se deberían extraer al menos un issue. Lo suyo sería que cerraras los issues incorrectos con una etiqueta "wontfix", "cerrándolos" desde un mensaje de commit en el que _borres_ el código creado hasta el momento. A continuación, tienes que crear diferentes issues _todos los cuales deben enlazar la HU_ que planteen los diferentes problemas existentes. Lo mejor sería que no hicieras nada hasta que escribas estos issues. Si vuelves a hacer código a partir de issues incorrectos, no será evaluable porque no se podrá saber si el código aporta valor o no. En todo caso, y a menos que @AlvaroSanPal95 vaya a continuar trabajando en tu repositorio, dado que tienes que empezar de nuevo yo diría que trabajaras sobre tu propio repositorio, porque cuando llegues al objetivo 4 vas a necesitar el código del objetivo 2, y si no lo han hecho, vas a tener que hacerlo tú desde cero. Así que si @AlvaroSanPal95 no lo soluciona prácticamente en un día, yo te diría que consideraras la opción de trabajar sobre tu propio repo.

Okay, pues seguiré trabajando sobre mi propio repositorio.

JJ commented:
Tendrías que cerrar el PR existente y trabajar sobre uno nuevo, porque al estar este en el repo de Álvaro, no puedes modificar nada.
